File: Simple_Linear_Regression.R
# Simple Linear Regression

#Importing data set
dataset = read_csv("Salary_Data.csv")

#Splitting dataset into training and test
#install.packages('caTools')
library(caTools)
#to get same results we set seed
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3) #ration is for training set
split
training_set = subset(dataset, split==TRUE)
test_set = subset(dataset, split==FALSE)

#Fitting simple LR to the Training set
regreesor = lm(formula = Salary~YearsExperience, data = training_set)
summary(regreesor)

#Predicting the test set results
y_pred = predict(regreesor, newdata = test_set)

#Visualising the Training set results
library(ggplot2) #to automate library
#Steps
# 1. plot all the observations in training data set
# 2. plot the regression line - predicted values
# 3. set title
# 4. set X and Y label
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regreesor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

#Visualising the Test set results
library(ggplot2) #to automate library

ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regreesor, newdata = training_set)), # dont need to replace with test
            #set becoz this prediction results are obtained from same unique reg line so ww will get the same reg line
            colour = 'blue') +
  ggtitle('Salary vs Experience (Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary')

File: Simple_Linear_Regression.py
# Simple Linear Regression

# Importing the libraries

#numpy library to include mathematical tools
import numpy as np
import matplotlib.pyplot as plt 
import pandas as pd 

#Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
#Creating matrix of features
X = dataset.iloc[:, :-1].values 
Y = dataset.iloc[:, 1].values 

#Splitting the dataset into the Training and Test set
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

#Fitting simple linear reg to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

#Predicting the Test set results
y_pred = regressor.predict(X_test)

# Visualising the Training set results
plt.scatter(X_train, Y_train, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

# Visualising the Test set results
plt.scatter(X_test, Y_test, color = 'red')
plt.plot(X_train, regressor.predict(X_train), color = 'blue')
plt.title('Salary vs Experience (Test set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()
